crates/router/src/connector/worldpay/response.rs

```diff
@@ -20,13 +21,13 @@ pub enum WorldpayPaymentResponseFields {
     DDCResponse(DDCResponse),
     FraudHighRisk(FraudHighRiskResponse),
     RefusedResponse(RefusedResponse),
+    ThreeDsChallenged(ThreeDsChallengedResponse),
 }
 
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct AuthorizedResponse {
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub payment_instrument: Option<PaymentsResPaymentInstrument>,
+    pub payment_instrument: PaymentsResPaymentInstrument,
```

Comment:
Why is this change required?

Reply:
Two reasons

- AuthorizedResponse had every field as optional, so any of the responses (DDC / refusal / fraud) would've deserialized into the AuthorizedResponse variant
- AuthorizedResponse will always have the details of which payment instrument was used, so making it non optional, also making the payment_instrument.type as non optional